include/dpp/cluster.h

```diff
@@ -3053,6 +3053,22 @@ class DPP_EXPORT cluster {
 	 */
 	void current_user_leave_guild(snowflake guild_id, command_completion_event_t callback = utility::log_error());
 
+	/**
+	 * @brief Create a thread in forum channel
+	 * @note This method supports audit log reasons set by the cluster::set_audit_reason() method.
+	 *
+	 * @see https://discord.com/developers/docs/resources/channel#start-thread-in-forum-channel
+	 * @param thread_name Name of the forum thread
+	 * @param channel_id Forum channel in which thread to create
+	 * @param msg The message to start the thread with
+	 * @param auto_archive_duration Duration in minutes to automatically archive the thread after recent activity. Can be set to: `60`, `1440` (default), `4320`, `10080`
+	 * @param rate_limit_per_user amount of seconds a user has to wait before sending another message (0-21600); bots, as well as users with the permission manage_messages, manage_thread, or manage_channel, are unaffected
+	 * @param applied_tags List of IDs of forum tags (dpp::forum_tag) to apply to this thread
+	 * @param callback Function to call when the API call completes.
+	 * On success the callback will contain a dpp::thread object in confirmation_callback_t::value. On failure, the value is undefined and confirmation_callback_t::is_error() method will return true. You can obtain full error details with confirmation_callback_t::get_error().
+	 */
+	void thread_create_in_forum(const std::string& thread_name, snowflake channel_id, message& msg, uint16_t auto_archive_duration = 1440, uint16_t rate_limit_per_user = 0, std::vector<snowflake> applied_tags = {}, command_completion_event_t callback = utility::log_error());
```

Comment:
should we make the `auto_archive_duration` param to the new type `auto_archive_duration_t`?

Reply:
yeah then you can't put an invalid value in it

Reply:
the downside is that we have to adapt the value to minutes in the method, something like:
```c++
if (auto_archive_duration == arc_1_hour) {
		j["auto_archive_duration"] = 60;
	} else if (auto_archive_duration == arc_1_day) {
		j["auto_archive_duration"] = 1440;
	} else if (auto_archive_duration == arc_3_days) {
		j["auto_archive_duration"] = 4320;
	} else if (auto_archive_duration == arc_1_week) {
		j["auto_archive_duration"] = 10080;
	}
```

Reply:
whats wrong with `switch` and `case`?

Reply:
maybe someone wants to store the auto_archive_duration in their database, and using an anum makes it complicated idk. but maybe im thinking too much and this would be an edge case.

Reply:
yeah i think youre over thinking that :D

Reply:
yeah, ill change that to the enum

Reply:
done